### PR TITLE
Clean up test helper module

### PR DESCRIPTION
**Pull request by OldSneerJaw**
Added test helper documentation to exported properties/functions and to the README and refactored its functions to eliminate superfluous parameters and add more conveniences. Updated existing test cases to make use of the new conveniences. Also created test cases specifically for verifying all the channels that are assigned to a document.

### CODE REVIEW

File: etc/test-helper.js [hunk 1]
```diff
@@ -46,16 +46,16 @@ function verifyDocumentAccepted(doc, oldDoc, expectedChannels) {
   }
 }
 
-function verifyDocumentCreated(doc, oldDoc) {
-  verifyDocumentAccepted(doc, oldDoc, 'add');
+function verifyDocumentCreated(doc, expectedChannels) {
```
Comment (OldSneerJaw) [inline etc/test-helper.js]: For the sake of convenience, there's no need for the caller to include the old document when creating a new document.

File: etc/test-helper.js [hunk 2]
```diff
@@ -46,16 +46,16 @@ function verifyDocumentAccepted(doc, oldDoc, expectedChannels) {
...
 
-function verifyDocumentReplaced(doc, oldDoc) {
-  verifyDocumentAccepted(doc, oldDoc, 'replace');
+function verifyDocumentReplaced(doc, oldDoc, expectedChannels) {
+  verifyDocumentAccepted(doc, oldDoc, expectedChannels || 'replace');
 }
 
-function verifyDocumentDeleted(doc, oldDoc) {
-  verifyDocumentAccepted(doc, oldDoc, 'remove');
+function verifyDocumentDeleted(oldDoc, expectedChannels) {
```
Comment (OldSneerJaw) [inline etc/test-helper.js]: There's no longer a need for the caller to include the updated document when deleting an existing document.

File: test/general-spec.js [hunk 3]
```diff
@@ -81,8 +82,50 @@ describe('Functionality that is common to all documents:', function() {
...
-      expect(requireAccess.callCount).to.equal(1);
-      expect(requireAccess.calls[0].arg).to.eql('remove');
+      expect(requireAccess.callCount).to.be(1);
+      expect(requireAccess.calls[0].arg.length).to.be(2);
+      expect(requireAccess.calls[0].arg).to.contain('delete');
+      expect(requireAccess.calls[0].arg).to.contain('remove');
+    });
+  });
+
+  describe('channel assignment', function() {
```
Comment (OldSneerJaw) [inline test/general-spec.js]: Verifies the channels that are assigned to a document via the `channel` function, which are a superset of the channels that are required to perform each operation (i.e. that are passed to the `requireAccess` function).

File: README.md [hunk 4]
```diff
@@ -337,3 +337,65 @@ Validation for complex data types, which allow for nesting of child properties a
...
+
+After that, run the `make-sync-function` script on your document definitions to generate the sync function to test. For example:
+
+    `node_modules/synctos/make-sync-function /path/to/my-doc-definitions.js /path/to/my-generated-sync-function.js`
+
+Next, create a new spec file in your project's `test/` directory (e.g. `test/foobar-spec.js`) and import the test helper module into the empty spec:
+
+    const testHelper = require('../node_modules/synctos/etc/test-helper.js');
+
+Create a new `describe` block to encapsulate the forthcoming test cases and and also initialize the synctos test helper before each test case using the `beforeEach` function. For example:
```
Comment (ziemek) [inline README.md]: double `and`.

**Comment by ziemek** [comment]
Looks good, nice documentation and improvements! 👍

**Pull request merged**